app/src/main/java/com/example/notebook/view/fragments/my_notes/creation_note/CreationNoteViewModel.kt

```diff
@@ -0,0 +1,22 @@
+package com.example.notebook.view.fragments.my_notes.creation_note
+
+import androidx.lifecycle.ViewModel
+import androidx.lifecycle.viewModelScope
+import com.example.notebook.domain.models.Note
+import com.example.notebook.domain.use_cases.AddNoteUseCase
+import dagger.hilt.android.lifecycle.HiltViewModel
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.launch
+import javax.inject.Inject
+
+@HiltViewModel
+class CreationNoteViewModel @Inject constructor(
+    private val addNoteUseCase: AddNoteUseCase
+): ViewModel() {
+
+    fun addNote(note: Note) {
+        viewModelScope.launch(Dispatchers.IO) {
```

Comment:
I would rather not specify the dispatcher type here and switch the context in the repo or use case. This way would be sure that operation occurs on the Dispatcher you expect it to execute